### PR TITLE
[19669] Added idl file to check constants.

### PR DESCRIPTION
**Pull request by adriancampo**
Added idl file to check constants.

### CODE REVIEW

File: IDL/constants.idl
Comment (JLBuenoLopez) [inline IDL/constants.idl:1]: IDL v4.2 syntax rule (6) establishes that:

```
(6)    <const_type>              ::= <integer_type>
                                              | <floating_pt_type>
                                              | <fixed_pt_const_type>
                                              | <char_type>
                                              | <wide_char_type>
                                              | <boolean_type>
                                              | <octet_type>
                                              | <string_type>
                                              | <wide_string_type>
                                              | <scoped_name>
```

I am missing this last case. Using an alias of one of the previous types to define a constant.

**Review by JLBuenoLopez — CHANGES_REQUESTED** [review]

File: IDL/constants.idl
Comment (JLBuenoLopez) [reply IDL/constants.idl:1]: It would be nice to also check if a constant defined in other IDL file can be used in this IDL.

Comment (adriancampo) [reply IDL/constants.idl:1]: Added in [64813b1](https://github.com/eProsima/dds-types-test/pull/8/commits/64813b100f906c17238ad9e5ee29aa8d435b57c9)

File: IDL/constants.idl [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        short module1_array_literal_const_moduled[const_moduled];
+        short module1_array_literal_const_alias_const_moduled[alias_const_moduled];
+    };
+};
+
+module const_module2
+{
+    const short const_moduled = 22;
+    typedef short alias_short_moduled;
+    const alias_short_moduled alias_const_moduled = 2;
```
Comment (JLBuenoLopez) [inline IDL/constants.idl]:
```suggestion
    const const_moduled::alias_short_moduled alias_const_moduled = 2;

```

**Review by JLBuenoLopez — CHANGES_REQUESTED** [review]

**Review by JLBuenoLopez — DISMISSED** [review]
LGTM. Not ready for merge until the constants are parsed correctly in Fast DDS-Gen

File: IDL/constants.idl [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+{
+    short array_literal_const_short[const_short];
+    unsigned short array_literal_const_ushort[const_ushort];
+    long array_literal_const_long[const_long];
+    unsigned long array_literal_const_ulong[const_ulong];
+    long long array_literal_const_longlong[const_longlong];
+    unsigned long long array_literal_const_ulonglong[const_ulonglong];
+    float array_literal_const_float[const_float];
+    double array_literal_const_double[const_double];
+    long double array_literal_const_longdouble[const_longdouble];
```
Comment (richiware) [inline IDL/constants.idl]: No sense having floating-points as array's indexes. Remove these cases please.

File: IDL/constants.idl [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    int8 array_literal_const_int8[const_int8];
+    uint8 array_literal_const_uint8[const_uint8];
+    int16 array_literal_const_int16[const_int16];
+    uint16 array_literal_const_uint16[const_uint16];
+    int32 array_literal_const_int32[const_int32];
+    uint32 array_literal_const_uint32[const_uint32];
+    int64 array_literal_const_int64[const_int64];
+    uint64 array_literal_const_uint64[const_uint64];
+    string array_literal_const_string[const_string];
+    wstring array_literal_const_wstring[const_wstring];
```
Comment (richiware) [inline IDL/constants.idl]: No sense having strings as array's indexes. Remove these cases please.

**Review by richiware — CHANGES_REQUESTED** [review]
Please, test array's indexes as operations: `array[const + const]`, `array[ const * const]`, so on...

File: IDL/constants.idl [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+module const_module2
+{
+    const short const_moduled = 22;
+    const const_module1::alias_short_moduled alias_const_moduled = 2;
+
+    struct Module2ConstsLiteralsStruct
+    {
+        short module2_array_literal_const_moduled[const_moduled];
+        short module2_array_literal_const_alias_const_moduled[alias_const_moduled];
```
Comment (richiware) [inline IDL/constants.idl:52]: Please, add theses cases in this Module2ConstLiteralsStruct structure:

```omg-idl
        short module2_array_literal_const_scoped_moduled[const_module2::const_moduled];
        short module2_array_literal_module1_const_moduled[const_module1::const_moduled];
```

**Review by richiware — CHANGES_REQUESTED** [review]

File: IDL/constants.idl [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    unsigned long array_literal_const_ulong[const_ulong];
+    long long array_literal_const_longlong[const_longlong];
+    unsigned long long array_literal_const_ulonglong[const_ulonglong];
+    float array_literal_const_float[const_float];
+    double array_literal_const_double[const_double];
+    long double array_literal_const_longdouble[const_longdouble];
+    boolean array_literal_const_boolean[const_boolean];
+    octet array_literal_const_octet[const_octet];
+    char array_literal_const_char8[const_char8];
+    wchar array_literal_const_char16[const_char16];
```
Comment (richiware) [inline IDL/constants.idl]: Also these two not have sense.

**Review by richiware — CHANGES_REQUESTED** [review]

File: IDL/constants.idl (hunk 5 above)
Comment (adriancampo) [reply IDL/constants.idl]: removed in [3656b48](https://github.com/eProsima/dds-types-test/pull/8/commits/3656b48bb0b974a4ad582a8ac93cb33cd0727ea7)

File: IDL/constants.idl (hunk 4 above)
Comment (adriancampo) [reply IDL/constants.idl:52]: added in [3656b48](https://github.com/eProsima/dds-types-test/pull/8/commits/3656b48bb0b974a4ad582a8ac93cb33cd0727ea7)

File: IDL/constants.idl (hunk 3 above)
Comment (adriancampo) [reply IDL/constants.idl]: removed in [3656b48](https://github.com/eProsima/dds-types-test/pull/8/commits/3656b48bb0b974a4ad582a8ac93cb33cd0727ea7)

File: IDL/constants.idl (hunk 2 above)
Comment (adriancampo) [reply IDL/constants.idl]: removed in [3656b48](https://github.com/eProsima/dds-types-test/pull/8/commits/3656b48bb0b974a4ad582a8ac93cb33cd0727ea7)

**Review by richiware — CHANGES_REQUESTED** [review]
Compilation error due array size is 0
```
 In file included from /home/ricardo/workspace/eprosima/fastddsgen/build/test/integration/idls/constants/thirdparty/dds-types-test/IDL/constants.cxx:29:
    /home/ricardo/workspace/eprosima/fastddsgen/build/test/integration/idls/constants/thirdparty/dds-types-test/IDL/constants.h:1243:120: error: initializer for 'std::__array_traits<short int, 0>::_Type' must be brace-enclosed
     1243 |     std::array<int16_t, const_module2::const_moduled%const_module1::const_moduled> m_array_literals_operations8_const{0};
```

**Review by richiware — APPROVED** [review]

**Pull request merged**